### DISCUSSION TITLE
How can I change the line-wrapping tag of code block?

### DISCUSSION THREAD

**Discussion by HeekangPark** [nuxt/content#1972] [Q&A] [answered]
It seems that Shiki parses the code block's content, and conducts syntax highlighting, while wrapping each line with `<div class="line">` tags.

Here's my question.

1. Is there any method to change the line-wrapping tags? E.g., `<div>` to `<span>`?
2. Is there any method to change the class of line-wrapping tags? E.g., adding line index(`<div class="line line-13">`, for 13th line, for example)?

**Comment by nobkd** [comment, marked as answer]
1. Before, `<span>` tags were used to generate the lines, but they were replaced with `div`s for highlighted code blocks because of line breaks missing in Firefox when copying code by selection.
  See #1883 and #1885.
2. A line number class should be fairly easy to implement, as the line index is already used to set the line as highlighted.
  See [this code](https://github.com/nuxt/content/blob/884f5d8260dd4c297abaa87bb6a60a355e03978d/src/runtime/transformers/shiki/highlighter.ts#L147).

I'll try to do a PR: #1973

**Reply by nobkd** [reply]
Oh, also, if you just want to highlight lines by default for one code block, you might want to look into this: https://content.nuxtjs.org/api/components/prose#prosecode
As you might see, the lines mentioned in `{}` get a `highlight` class
